src/ts/replayprotection/bitflip.ts

```diff
@@ -50,22 +53,21 @@ export class BitFlip extends ReplayProtectionAuthority {
     for (let i = 0; i < 1000; i) {
       try {
         // Try to find an empty bit
-        bitToFlip = this.findEmptyBit(bitmap);
+        bitToFlip = this.findEmptyBit(this.bitmap);
 
         // Did we find one?
         if (bitToFlip.eq(new BigNumber("-1"))) {
           // No, let's try the next bitmap
-          index = index.add(1);
-          bitmap = await this.accessNonceStore(signer, index, this.contract);
+          this.index = this.index.add(1);
+          this.bitmap = await this.accessNonceStore(this.index);
         } else {
           // We found an empty bit
           foundEmptyBit = true;
 
-          // Keep track of index and new flipped bitmap
-          this.indexTracker.set(signer.address, index);
-          const flipped = this.flipBit(bitmap, bitToFlip);
-          this.bitmapTracker.set(signer.address, flipped);
-          return { index, bitToFlip };
+          const flipped = this.flipBit(this.bitmap, bitToFlip);
+          this.bitmap = flipped;
+          const newIndex = this.index;
+          return { newIndex, bitToFlip };
         }
       } catch (e) {
```

Comment:
its to protect from infura exceptions. if you are checking the nonce store every 500 ms, you could end up getting rate limited. so the catch will just wait() and then move onto the next nonce.

Reply:
Your code isnt doing that though, you're swallowing ALL errors in this.accessNonceStore so actually any error from infura will never be caught in this catch.

Even it were being caught in this catch, you shouldnt handle errors in this way, you need to classify errors, especially if you're looking for a specific error, to avoid hiding other bugs in your code.

Your code should be useable in contexts other than infura, if you need infura specific throttling/error handling you should inject this behaviour. On top of this when infura decides to throttle you they provide you with a backoff time period - you wont be able to make any requests until then, so at the moment your code will just be unresponsive waiting until that time period is over - which could be minutes or hours.

Reply:
I've removed the try and catch. I've put a small wait(100) and the loop to 30. If it fails after that it'll throw an exception.

We should make an issue that works on a better strategy for finding the next bit to flip. e.g. right now it'll just pick a random number then try the next sequential 30 numbers of that.